multiaddr_test.go

```diff
@@ -582,11 +582,11 @@ func TestBinaryMarshaler(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	addr2 := newMultiaddr(t, "")
+	var addr2 multiaddr
```

Comment:
This is the issue I was talking about. This could be done by manually calling `NewMultiaddr` before but now one needs to construct a _valid_ multiaddr.

Reply:
but would work by creating a non-empty multiaddress (which just will be replaced) ?

Reply:
Yes, I guess. Although... multiaddrs are really supposed to be immutable. I guess this is just an unavoidable edge-case.